File: Javascript/ValidateStackSequences.js
// Leetcode Problem - 946

var validateStackSequences = function (pushed, popped) {
  const stack = [];
  let poppedIndex = 0;

  for (const element of pushed) {
    stack.push(element);

    while (
      poppedIndex < popped.length &&
      stack.length > 0 &&
      popped[poppedIndex] === stack[stack.length - 1]
    ) {
        stack.pop()
        poppedIndex+=1
    }
  }

  return stack.length > 0 ? false : true
};

File: Python/ValidateStackSequences.py
## Leetcode Problem - 946

def validateStackSequences(pushed,popped):
    poppedIndex = 0
    stack = []

    for element in pushed:
        stack.append(element)

        while poppedIndex<len(popped) and stack and popped[poppedIndex] == stack[-1]:
            stack.pop()
            poppedIndex+=1

    return not stack

    

    



pushed = [1,2,3,4,5]
popped = [4,5,3,2,1]

print(validateStackSequences(pushed,popped))